boards: xtensa: Add support for Lolin S2 Mini

mfischer opened the pull request:
Add support for LOLIN S2 Mini board, a cheap board also sold as Wemos S2 Mini.

mfischer committed:
boards: xtensa: Add support for Lolin S2 Mini

Add support for LOLIN S2 Mini board, a cheap board
also sold as Wemos S2 Mini.

Signed-off-by: Moritz Fischer <moritzf@google.com>

sylvioalves commented on boards/xtensa/esp32s2_lolin_mini/esp32s2_lolin_mini_defconfig:
```diff
@@ -0,0 +1,24 @@
+CONFIG_SOC_ESP32S2=y
+CONFIG_MAIN_STACK_SIZE=2048
+
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=240000000
```
You can keep this, but it is no required.

mfischer replied:
Can I remove all of them or just line 9? Thanks!

sylvioalves replied:
Just line 9.

mfischer replied:
Done

mfischer commented on boards/xtensa/esp32s2_lolin_mini/esp32s2_lolin_mini.yaml:
```diff
@@ -0,0 +1,16 @@
+toolchain:
+  - zephyr
+supported:
+  - adc
```
So I was thinking about that some more. Ultimately the board has very little fixed functionality beyond a gpio-button and a gpio-led.

I have several samples I made work, I was thinking if your option 1 from above is the correct approach, for example the UART can be muxed to just about any pin IIUC.

Should I just remove the UART stuff from the board and add them to the samples such as `samples/hello_world` or `samples/subsys/canbus/isotp` via a board overlay in each case?

Is it fine for a board to 'pick' a set of pins for say UART? Someone might want to use the board without UART and use the pins for something else. They could remove them obviously via /delete-node/ but maybe they shouldn't be part of the board dts in the first place then.

Opinions?

mfischer replied:
Done?

MaureenHelm commented:
@sylvioalves ping

mfischer commented:
@sylvioalves it seems the failing test fails also with `west build -p -b esp32s2_saola -T tests/lib/heap/libraries.heap` ideas?

mfischer commented:
> @sylvioalves it seems the failing test fails also with `west build -p -b esp32s2_saola -T tests/lib/heap/libraries.heap` ideas?

Hrmmm ... Investigation showed this test was disabled for that board, let me do the same :)

marekmatej commented on boards/xtensa/esp32s2_lolin_mini/esp32s2_lolin_mini-pinctrl.dtsi:
```diff
@@ -0,0 +1,22 @@
+
+	uart0_default: uart0_default {
+		group1 {
+			pinmux = <UART0_TX_GPIO16>;
```
you can add the `output-high;` attribute to prevent uart to produce garbage before is properly configured.

mfischer replied:
@sylvioalves ping. Do you want me to fix this first?

mfischer replied:
Fixed. PTAL

mfischer commented:
@marekmatej @sylvioalves ping :)

marekmatej commented:
Please note upcoming changes in boards/soc Kconfig https://github.com/zephyrproject-rtos/zephyr/pull/58454

mfischer commented:
> Please note upcoming changes in boards/soc Kconfig #58454

Should we wait for this or can this go in before?

marekmatej commented:
> > Please note upcoming changes in boards/soc Kconfig #58454
> 
> Should we wait for this or can this go in before?

If this is merged prior to SOC refactor, I'd need to update your work afterward. So I'd prefer if you can wait a little more.

mfischer commented:
> dev-review: @mfischer - #58454 is merged, please update this if necessary @sylvioalves - re-ping

Rebased and updated. `samples/basic/blinky` and `samples/hello_world` still work

MaureenHelm commented:
@sylvioalves please revisit

marekmatej commented on boards/xtensa/esp32s2_lolin_mini/doc/index.rst:
```diff
@@ -0,0 +1,92 @@
+.. _esp32s2_lolin_mini:
+
+ESP32-S2
```
Maybe change the name to "esp32s2 lolin mini"

mfischer replied:
Done

kartben commented on boards/xtensa/esp32s2_lolin_mini/doc/index.rst:
```diff
@@ -0,0 +1,92 @@
+
+ESP32-S2 is a highly integrated, low-power, single-core Wi-Fi Microcontroller SoC, designed to be secure and
+cost-effective, with a high performance and a rich set of IO capabilities. [1]_
+
```
While I don't think it's 100% mandatory, we tend to include a photo for each board.

marekmatej replied:
I was going to ask the same :)

mfischer replied:
Done.

MaureenHelm commented:
@mfischer can you please have a look at the review comments?

sylvioalves commented:
LGTM. Just waiting previous comments to be addressed.

kartben commented on boards/xtensa/esp32s2_lolin_mini/doc/img/esp32_s2_lolin_mini.jpg:
This picture is pretty blurry, maybe you could use a resized version of https://www.wemos.cc/en/latest/_static/boards/s2_mini_v1.0.0_1_16x16.jpg and just credit Wemo in the figure caption.
You may also want to convert the image to webp, as it is typically better than jpg at compressing.

mfischer replied:
This is more a function of my poor photography / picture editing skills. Can we do this in a follow up?

sylvioalves replied:
@mfischer, I just saw this comment (sorry, was on leave). I guess updating the image in this PR would make more sense. Any reasons not to?

mfischer commented:
@sylvioalves ping
